pkg/volume/util/util.go

```diff
@@ -282,10 +282,12 @@ func checkVolumeNodeAffinity(pv *v1.PersistentVolume, nodeLabels map[string]stri
 			if err != nil {
 				return fmt.Errorf("Failed to parse MatchExpressions: %v", err)
 			}
-			if !selector.Matches(labels.Set(nodeLabels)) {
-				return fmt.Errorf("NodeSelectorTerm %+v does not match node labels", term.MatchExpressions)
```

Comment:
This needs to be updated when MatchFields is added.  Consider using a common validate NodeSelectorTerms function instead.

Reply:
Ok one doesn't exist yet.  For the current state, I'll keep it as is, and can cherry pick this to 1.10.

Then when MatchFields is added in 1.11, this function can be updated as well to use a common method.  cc @k82cn

Reply:
@msau42 @k82cn - I’m fine with this, but I believe we are ANDing for nodeAffinity in the scheduler. This might confuse users.

Reply:
@liggitt pointed out to to me that the scheduler is ORing [here](https://github.com/kubernetes/kubernetes/blob/HEAD/pkg/scheduler/algorithm/predicates/predicates.go#L782-L784), and that is also how it's described in the docs.

Reply:
Oh indeed that’s nodeSelector and nodeaffinity that are ANDed.

Reply:
> Then when MatchFields is added in 1.11, this function can be updated as well to use a common method.

Sure, I'll add a helper func for volume and scheduler :)